client-core/src/synchronizer.rs

```diff
@@ -45,30 +49,37 @@ impl PollingSynchronizer {
         let wallets = self.wallets.clone();
         let progress = self.progress.clone();
 
+        self.should_run.store(true, Ordering::Relaxed);
+        let should_run = self.should_run.clone();
+
         let (sender, receiver) = channel();
-        thread::spawn(move || loop {
-            let wallets_to_synchronize = wallets
-                .lock()
-                .expect("Unable to acquire lock on wallets to synchronize in polling synchronizer")
-                .clone();
-
-            for (name, enckey) in wallets_to_synchronize.iter() {
-                let result = WalletSyncer::with_obfuscation_config(
-                    config.clone(),
-                    Some(sender.clone()),
-                    name.clone(),
-                    enckey.clone(),
-                )
-                .and_then(|syncer| syncer.sync());
-                if let Err(e) = result {
-                    log::error!("Error while synchronizing wallet [{}]: {:?}", name, e);
+        self.sync_thread = Some(thread::spawn(move || {
+            while should_run.load(Ordering::Relaxed) {
+                let wallets_to_synchronize = wallets
+                    .lock()
+                    .expect(
+                        "Unable to acquire lock on wallets to synchronize in polling synchronizer",
+                    )
+                    .clone();
+
+                for (name, enckey) in wallets_to_synchronize.iter() {
+                    let result = WalletSyncer::with_obfuscation_config(
+                        config.clone(),
+                        Some(sender.clone()),
+                        name.clone(),
+                        enckey.clone(),
+                    )
+                    .and_then(|syncer| syncer.sync());
```

Comment:
in testnet, this sync will take many hours.
maybe days for slow network

so i think for a few wallets,
each thread for wallet ?

or one block data, multiplexing,
but that's beyond this PR

Reply:
It's already recorded in #1060